### PR TITLE
Add connection_string to interface and use in interface_microservice

### LINKED ISSUE #1169: Add additional connect details to interface connect errors
For Ethernet connections this could IP/host & port info

### PR DESCRIPTION
**Pull request by jmthomas**
Still need to do Python but this is what I'm thinking

closes #1169

### CODE REVIEW

File: openc3/lib/openc3/interfaces/http_server_interface.rb [hunk 1]
```diff
@@ -31,6 +31,10 @@ def initialize(port = 80)
       @request_queue = Queue.new
     end
 
+    def connection_string
+      return "listening on #{@port}"
```
Comment (ryanmelt) [inline openc3/lib/openc3/interfaces/http_server_interface.rb:35]: Needs hostname too.

File: openc3/lib/openc3/microservices/interface_microservice.rb [hunk 2]
```diff
@@ -627,7 +627,7 @@ def handle_connection_lost(err = nil, reconnect: true)
     end
 
     def connect
-      @logger.info "#{@interface.name}: Connecting ..."
+      @logger.info "#{@interface.name}: Connection #{@interface.connection_string}"
```
Comment (ryanmelt) [inline openc3/lib/openc3/microservices/interface_microservice.rb:630]: Connection doesn't read as nice as connecting...   Maybe just Connecting: ?

**Review by ryanmelt — CHANGES_REQUESTED** [review]

File: openc3/lib/openc3/microservices/interface_microservice.rb (hunk 2 above)
Comment (jmthomas) [reply openc3/lib/openc3/microservices/interface_microservice.rb:630]: How about `Connect`

File: openc3/lib/openc3/interfaces/http_server_interface.rb (hunk 1 above)
Comment (jmthomas) [reply openc3/lib/openc3/interfaces/http_server_interface.rb:35]: Hostname isn't part of the connection parameters. It's just: `@server = WEBrick::HTTPServer.new :Port => @port`

Comment (ryanmelt) [reply openc3/lib/openc3/interfaces/http_server_interface.rb:35]: It should be though.  Should work the same as tcpip_server_interface.

File: openc3/lib/openc3/interfaces/tcpip_server_interface.rb [hunk 3]
```diff
@@ -119,6 +118,16 @@ def initialize(write_port,
       @connected = false
     end
 
+    def connection_string
+      if @write_port == @read_port
+        return "listening on #{@write_port} (R/W)"
```
Comment (ryanmelt) [inline openc3/lib/openc3/interfaces/tcpip_server_interface.rb:123]: Should include hostname

**Review by ryanmelt — CHANGES_REQUESTED** [review]

**Pull request merged**